docs/06-autenticacao-seguranca.md

```diff
@@ -0,0 +1,506 @@
+        await userProfileStore.save(session.user);
+        return Right(session);
+      },
+    );
+  }
+}
+```
+
+### 2. Modo Camuflado (Stealth Mode)
+
+O modo camuflado é uma funcionalidade crítica de segurança:
+
+```dart
+class AuthenticateStealthUserUseCase {
+  Future<Either<Failure, SessionEntity>> call({
+    required String pin,
+  }) async {
+    // Hash do PIN para comparação
+    final hashedPin = _hashPin(pin);
+    
+    // Verificar PIN offline primeiro
+    final storedHash = await appConfiguration.offlineHash;
+    
+    if (storedHash.isNotEmpty && hashedPin == storedHash) {
+      // Login offline bem-sucedido
+      return _offlineLogin();
+    }
+    
+    // Tentar login online
+    return _onlineStealthLogin(hashedPin);
+  }
+  
+  String _hashPin(String pin) {
+    // Usar crypt para hash seguro
+    final salt = '\$2a\$10\$' + _generateSalt();
+    return Crypt.sha256(pin, salt: salt).toString();
+  }
+}
+```
+
+#### Características do Modo Camuflado:
+- Interface disfarçada de calculadora
+- PIN ao invés de senha completa
+- Dados separados e criptografados
+- Logout automático por inatividade
+- Sem rastros visíveis do app real
+
+### 3. Login Anônimo
+
+Para acesso rápido a recursos essenciais:
+
+```dart
+class AuthenticateAnonymousUserUseCase {
+  Future<Either<Failure, SessionEntity>> call() async {
+    // Gerar identificador único temporário
+    final tempId = Uuid().v4();
+    
+    // Criar sessão anônima limitada
+    final anonymousSession = SessionEntity(
+      token: 'anonymous_$tempId',
+      user: UserProfile.anonymous(),
+      permissions: ['read_public', 'emergency_features'],
+    );
+    
+    await appConfiguration.saveAnonymousSession(anonymousSession);
+    
+    return Right(anonymousSession);
+  }
+}
+```
+
+## Armazenamento Seguro
+
+### 1. Flutter Secure Storage
+
+Para dados sensíveis como tokens e credenciais:
+
+```dart
+class SecureLocalStorage implements ISecureStorage {
+  static const _storage = FlutterSecureStorage();
+  
+  // Opções de segurança para Android
+  static const _androidOptions = AndroidOptions(
+    encryptedSharedPreferences: true,
+    sharedPreferencesName: 'secure_prefs',
+    preferencesKeyPrefix: 'secure_',
+  );
+  
+  // Opções de segurança para iOS
+  static const _iosOptions = IOSOptions(
+    accessibility: IOSAccessibility.first_unlock_this_device,
+    accountName: 'penhas_secure',
+  );
+  
+  Future<void> saveSecure(String key, String value) async {
+    await _storage.write(
+      key: key,
+      value: value,
+      aOptions: _androidOptions,
+      iOptions: _iosOptions,
+    );
+  }
+  
+  Future<String?> readSecure(String key) async {
+    return await _storage.read(
+      key: key,
+      aOptions: _androidOptions,
+      iOptions: _iosOptions,
+    );
+  }
+  
+  Future<void> deleteSecure(String key) async {
+    await _storage.delete(
+      key: key,
+      aOptions: _androidOptions,
+      iOptions: _iosOptions,
+    );
+  }
+}
+```
```

Comment:
```suggestion
```dart
class SecureLocalStorage implements ILocalStorage {
  factory SecureLocalStorage({
    FlutterSecureStorage storage = const FlutterSecureStorage(
      aOptions: AndroidOptions(resetOnError: true),
    ),
  }) =>
      SecureLocalStorage._(storage, {});

  const SecureLocalStorage._(this._storage, this._cache);

  final FlutterSecureStorage _storage;
  final Map<String, dynamic> _cache;
  }
```
```